js/modules/k6/http/request.go

```diff
@@ -84,6 +84,24 @@ func (c *Client) Request(method string, url goja.Value, args ...goja.Value) (*Re
 		}
 		return &Response{Response: r, client: c}, nil
 	}
+	if callback != nil {
+		f := c.moduleInstance.vu.Reserve()
```

Comment:
`f` is used twice further down the code, I would prefer if it had a more explicit name. Also, it's only used in the scope of the goroutine, was there an explicit reason to not declare inside it? 😸

Reply:
> Also, it's only used in the scope of the goroutine, was there an explicit reason to not declare inside it?

Literally what the `Reserve` documentaiton is trying to explain. You need to reserve on the event loop while executing on the event loop otherwise there is nothing stopping the event loop(and the iteration) to finish *before* you `Reserve` is executed.